runtime/slice.go

```diff
@@ -156,6 +156,58 @@ func initSliceType(map[string]*Object) {
 	SliceType.slots.Repr = &unaryOpSlot{sliceRepr}
 }
 
+func sliceIndex(f *Frame, o *Object) (i int64, raised *BaseException) {
+	if index := o.typ.slots.Index; index != nil {
+		o, raised = index.Fn(f, o)
+		if raised != nil {
+			return int64(0), raised
+		}
+	}
+	if o.isInstance(IntType) {
```

Comment:
Actually above index.Fn return the object and fall though, so they are needed.
Instead I wrote a comment instead, since it looks like confusion can come from reassigning variable `o`.